File: App Perpus Sederhana/Kategori.php
<?php
//Simpanlah dengan nama file : Kategori.php
require_once 'database.php';
class Kategori 
{
    private $db;
    private $table = 'kategori';
    public $kode_kategori = "";
    public $nama_kategori = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_kode_kategori(int $kode_kategori)
    {
        $query = "SELECT * FROM $this->table WHERE kode_kategori = $kode_kategori";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`kode_kategori`,`nama_kategori`) VALUES ('$this->kode_kategori','$this->nama_kategori')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET kode_kategori = '$this->kode_kategori', nama_kategori = '$this->nama_kategori' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_kode_kategori($kode_kategori): int
    {
        $query = "UPDATE $this->table SET kode_kategori = '$this->kode_kategori', nama_kategori = '$this->nama_kategori' 
        WHERE kode_kategori = $kode_kategori";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_kode_kategori($kode_kategori): int
    {
        $query = "DELETE FROM $this->table WHERE kode_kategori = $kode_kategori";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>

File: App Perpus Sederhana/Kategori.py
import requests
import json
class Kategori:
    def __init__(self):
        self.__id=None
        self.__kode_kategori = None
        self.__nama_kategori = None
        self.__url = "http://localhost/apperpus/kategori_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def kode_kategori(self):
        return self.__kode_kategori
        
    @kode_kategori.setter
    def kode_kategori(self, value):
        self.__kode_kategori = value
    @property
    def nama_kategori(self):
        return self.__nama_kategori
        
    @nama_kategori.setter
    def nama_kategori(self, value):
        self.__nama_kategori = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_kode_kategori(self, kode_kategori):
        url = self.__url+"?kode_kategori="+kode_kategori
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__kode_kategori = item['kode_kategori']
            self.__nama_kategori = item['nama_kategori']
        return data
    def simpan(self):
        payload = {
            "kode_kategori":self.__kode_kategori,
            "nama_kategori":self.__nama_kategori
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_kode_kategori(self, kode_kategori):
        url = self.__url+"?kode_kategori="+kode_kategori
        payload = {
            "kode_kategori":self.__kode_kategori,
            "nama_kategori":self.__nama_kategori
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_kode_kategori(self,kode_kategori):
        url = self.__url+"?kode_kategori="+kode_kategori
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text
